torch/csrc/autograd/variable.cpp

```diff
@@ -384,6 +384,20 @@ unsigned VariableHooks::_register_hook(const Tensor& self, std::function<Tensor(
 }
 
 void handle_view_on_rebase(DifferentiableViewMeta* diff_view_meta, bool indirect) {
+  // TODO: Remove this warning once we allow XLA to workaround CopySlices.
+  if (diff_view_meta->base_.device().type() == c10::DeviceType::XLA) {
+    std::string msg;
+    if (indirect) {
+      msg = "This view requires gradients but its base or another view of the same base has been modified inplace. ";
+    } else {
+      msg = "This view requires gradients and it's being modified inplace. ";
+    }
+    msg = c10::str(msg, "Backward through inplace update on view tensors is WIP for XLA backwend. "
+                   "Gradient might be wrong in certain cases. Running forward alone is fine. "
```

Comment:
this prints in the forward? It's not clear what "alone is fine" means.

Is there anyway to 'tag' this and only print the warning in the backward?

Putting an example of this triggering would be helpful to see the effect.

Reply:
Yes this prints in the forward when the inplace is done and only if inputs require gradients.
I think we don't want to delay the error as it would make it much harder for the user to find the faulty inplace.

Reply:
We generally don't warn in the forward if it will cause a problem in the backward, right?

This seems like a good place for a mechanism you can turn on that warns/errors in the forward and the error in the backward tells you to turn it on.

Reply:
Suggested rewording:

Running a backward pass through an inplace update on view tensors is a WIP for the XLA backend and may result in incorrect gradient computation in certain cases.  Note this warning is being triggered on the inplace update (not the corresponding backward pass), and this update is safe if a backward pass is not run.  To work around this limitation and to silence this warning, please replace the inplace operation by the corresponding out-of-place operation."

Reply:
Thanks @gchanan ! It has been updated in https://github.com/pytorch/pytorch/pull/35543.

Reply:
For these, we actually warn/error in the forward because we know that the backward will be wrong. Note that if you Tensor does not require gradients, then this won't be called. So people that do not use the autograd won't see it.